From 274a5fdb801e4aab7c55181429300f7084c8261c Mon Sep 17 00:00:00 2001
From: Amrita Raju <amrita.raju@intel.com>
Date: Sun, 3 Feb 2019 12:18:02 +0530
Subject: [PATCH 1/2] iwlwifi: Add changes to enable p2p

- Set wifi.direct.interface to correct value
- Add p2p configuration file
- Remove not needed configuration file

Tracked-On: None
Signed-off-by: Amrita Raju <amrita.raju@intel.com>
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 groups/wlan/iwlwifi/init.rc     | 6 +++---
 groups/wlan/iwlwifi/option.spec | 1 -
 groups/wlan/iwlwifi/product.mk  | 8 ++------
 3 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/groups/wlan/iwlwifi/init.rc b/groups/wlan/iwlwifi/init.rc
index f06bf52..3219c19 100755
--- a/groups/wlan/iwlwifi/init.rc
+++ b/groups/wlan/iwlwifi/init.rc
@@ -1,7 +1,7 @@
 on post-fs-data
-    chmod 0660 /data/misc/wifi/p2p_supplicant.conf
-	setprop wifi.interface wlan0
-	setprop wifi.direct.interface p2p0
+    setprop wifi.interface wlan0
+    setprop wifi.direct.interface p2p-dev-wlan0
+
     # create config WiFi NVM folder
     mkdir /oem_config/wlan 0770 wifi system
 {{#gpp}}
diff --git a/groups/wlan/iwlwifi/option.spec b/groups/wlan/iwlwifi/option.spec
index 94acfe0..db745d4 100755
--- a/groups/wlan/iwlwifi/option.spec
+++ b/groups/wlan/iwlwifi/option.spec
@@ -1,5 +1,4 @@
 [defaults]
-tdls_auto = false
 iwl_pnvm_hw = DEFAULT
 gpp = false
 softap_dualband_allow = false
diff --git a/groups/wlan/iwlwifi/product.mk b/groups/wlan/iwlwifi/product.mk
index 10de23f..de29607 100755
--- a/groups/wlan/iwlwifi/product.mk
+++ b/groups/wlan/iwlwifi/product.mk
@@ -1,7 +1,7 @@
 PRODUCT_PACKAGES += \
     hostapd \
     hostapd_cli \
-	wificond \
+    wificond \
     wifilogd \
     wpa_supplicant \
     wpa_cli \
@@ -19,12 +19,8 @@ PRODUCT_PACKAGES += \
 #copy iwlwifi wpa config files
 PRODUCT_COPY_FILES += \
         device/intel/common/wlan/wpa_supplicant-common.conf:vendor/etc/wifi/wpa_supplicant.conf \
-{{#tdls_auto}}
         device/intel/common/wlan/iwlwifi/wpa_supplicant_overlay.conf:vendor/etc/wifi/wpa_supplicant_overlay.conf \
-{{/tdls_auto}}
-{{^tdls_auto}}
-        device/intel/common/wlan/iwlwifi/wpa_supplicant_overlay_no_tdls.conf:vendor/etc/wifi/wpa_supplicant_overlay.conf \
-{{/tdls_auto}}
+        device/intel/common/wlan/iwlwifi/p2p_supplicant_overlay.conf:vendor/etc/wifi/p2p_supplicant_overlay.conf \
         frameworks/native/data/etc/android.hardware.wifi.xml:vendor/etc/permissions/android.hardware.wifi.xml
 
 {{#gpp}}
-- 
2.17.1

